libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -180,6 +180,54 @@ void NavEKF2_core::writeOptFlowMeas(uint8_t &rawFlowQuality, Vector2f &rawFlowRa
     }
 }
 
+// reset variances for mag body state
+void NavEKF2_core::resetMagBodyVariances(void)
+{
+    zeroCols(P,19,21);
+    zeroRows(P,19,21);
+    P[19][19] = sq(frontend->_magNoise);
+    P[20][20] = P[19][19];
+    P[21][21] = P[19][19];
+}
+
+// try changing to another compass
+void NavEKF2_core::tryChangeCompass(void)
```

Comment:
Just a nitpick, but to keep wording uniform throughout the code and to avoid confusion, I'd recommend renaming this function to `tryCompassSwitch()` or `trySwitchCompass()` since everywhere else we see flags like should_switch or the text below says "switching to compass %u" and the PR itself uses "switch" rather than "change." I think it's also more descriptive of the behavior since change can sometimes mean a replacement whereas switch usually implies that we are moving back and forth between existing hardware